src/coreclr/jit/morph.cpp

```diff
@@ -11701,7 +11701,12 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
             {
                 // Codegen for this instruction needs to be able to throw two exceptions:
                 fgAddCodeRef(compCurBB, bbThrowIndex(compCurBB), SCK_OVERFLOW);
```

Comment:
Add similar condition for the overflow exception as well? It should be needed only if the second operand is -1.

Reply:
Sure, makes sense.